### PR TITLE
remote-processor: Use a sized enum for message type

### PR DESCRIPTION
**Pull request by OznOg**
The former code was putting a enum into a uint8_t which may be wrong (depends
on the compiler taste).

This patch uses the c++11 ability to force the underlying int type of the enum
to make sure it fits a uint8.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/212%23issuecomment-139297783%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/212%23issuecomment-139501866%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/212%23issuecomment-139505062%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/212%23discussion_r39179214%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/212%23discussion_r39201901%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/212%23discussion_r39202382%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/212%23issuecomment-139297783%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-10T16%3A14%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-11T09%3A46%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-09-11T10%3A02%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2037de9f934846948a3fb2036234580a359cc5e2ad%20remote-processor/AnswerMessage.cpp%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/212%23discussion_r39179214%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20don%27t%20why%20it%20would%20be%20buggy...%20Currently%2C%20ESuccessAnswer%20is%20the%20only%20value%20that%20represents%20a%20success.%22%2C%20%22created_at%22%3A%20%222015-09-10T16%3A13%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22Me%20neither.%22%2C%20%22created_at%22%3A%20%222015-09-10T19%3A27%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/AnswerMessage.cpp%3AL59-67%22%7D%2C%20%22Pull%2037de9f934846948a3fb2036234580a359cc5e2ad%20remote-processor/AnswerMessage.cpp%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/212%23discussion_r39201901%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22line%202%20long.%20Anyway%20%60RemoteProtocolMsgType%60%20should%20be%20named%20%60MsgType%60%20in%20a%20%60remote%3A%3Aprotocol%60%20namespace.%5Cr%5CnOr%20even%20just%20%60Type%60%20as%20it%20is%20in%20the%20class%20%60Message%60.%5Cr%5CnMaybe%20do%20an%20other%20patch%20to%20commit%20the%20output%20of%20%60grep%20-rlz%20RemoteProtocolMsgType%20%7C%20xargs%20-0%20sed%20-i%20-e%20%27s/RemoteProtocolMsgType/Type%27%60%22%2C%20%22created_at%22%3A%20%222015-09-10T19%3A22%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20remote-processor/AnswerMessage.cpp%3AL35-44%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/dawagner%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/dawagner'></a>

- [ ] <a href='#crh-comment-Pull 37de9f934846948a3fb2036234580a359cc5e2ad remote-processor/AnswerMessage.cpp 16'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/212#discussion_r39179214'>File: remote-processor/AnswerMessage.cpp:L59-67</a></b>
- <a href='https://github.com/dawagner'></a> I don't why it would be buggy... Currently, ESuccessAnswer is the only value that represents a success.
- <a href='https://github.com/krocard'></a> Me neither.
- [ ] <a href='#crh-comment-Pull 37de9f934846948a3fb2036234580a359cc5e2ad remote-processor/AnswerMessage.cpp 6'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/212#discussion_r39201901'>File: remote-processor/AnswerMessage.cpp:L35-44</a></b>
- <a href='https://github.com/krocard'></a> line 2 long. Anyway `RemoteProtocolMsgType` should be named `MsgType` in a `remote::protocol` namespace.
Or even just `Type` as it is in the class `Message`.
Maybe do an other patch to commit the output of `grep -rlz RemoteProtocolMsgType | xargs -0 sed -i -e 's/RemoteProtocolMsgType/Type'`
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/212#issuecomment-139297783'>General Comment</a></b>
- <a href='https://github.com/dawagner'></a> :+1:
- <a href='https://github.com/dawagner'></a> :-1:

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/212?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/212?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/212?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/212'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/212'>Refresh</a>.<a href='#crh-update-none'></a>

File: remote-processor/AnswerMessage.cpp [hunk 1]
```diff
@@ -57,7 +59,8 @@ const string& CAnswerMessage::getAnswer() const
 // Status
 bool CAnswerMessage::success() const
 {
-    return getMsgId() == ESuccessAnswer;
+    /* FIXME this test is buggy as there are acutally 3 values in the enum... */
```
Comment (dawagner) [inline remote-processor/AnswerMessage.cpp]: I don't why it would be buggy... Currently, ESuccessAnswer is the only value that represents a success.

**Comment by dawagner** [comment]
:+1:

File: remote-processor/AnswerMessage.cpp [hunk 2]
```diff
@@ -35,7 +35,9 @@
 
 using std::string;
 
-CAnswerMessage::CAnswerMessage(const string& strAnswer, bool bSuccess) : base(bSuccess ? ESuccessAnswer : EFailureAnswer), _strAnswer(strAnswer)
+CAnswerMessage::CAnswerMessage(const string& strAnswer, bool bSuccess) :
+    base(bSuccess ? RemoteProtocolMsgType::ESuccessAnswer : RemoteProtocolMsgType::EFailureAnswer),
```
Comment (krocard) [inline remote-processor/AnswerMessage.cpp]: line 2 long

File: remote-processor/AnswerMessage.cpp (hunk 1 above)
Comment (krocard) [inline remote-processor/AnswerMessage.cpp]: Me neither.

**Comment by dawagner** [comment]
:-1:

**Comment by dawagner** [comment]
:shipit:

**Pull request merged**